core/src/main/java/org/commonjava/indy/core/content/DefaultDirectContentAccess.java

```diff
@@ -132,7 +133,7 @@ public Transfer retrieveRaw( final ArtifactStore store, String path, final Event
             throws IndyWorkflowException
     {
         // npm should handle the path as '/project' not '/project/package.json' when retrieves a remote registry
-        if ( store.getType() == remote && store.getPackageType().equals( NPM_PKG_KEY ) )
+        if ( store.getType() == remote && store.getPackageType().equals( NPM_PKG_KEY ) && path.endsWith( NPM_METADATA_NAME ))
         {
             String project = path.substring( 0, path.length()-13 );
```

Comment:
better use "/package.json".length() instead of 13 ?

Reply:
Done.